packages/components/src/icon-button/index.js

```diff
@@ -45,7 +45,7 @@ class IconButton extends Component {
 
 		let element = (
 			<Button aria-label={ label } { ...additionalProps } className={ classes }>
-				{ isString( icon ) ? <Dashicon icon={ icon } ariaPressed={ ariaPressed } /> : icon }
+				{ isString( icon ) ? <Dashicon icon={ icon } ariaPressed={ ariaPressed } { ...iconProps } /> : icon }
```

Comment:
Out of curiosity, since `icon` can already be specified as an element for more advanced usage, had you considered just passing your own `Dashicon` element directly, rather than introduce a new separate prop `iconProps`?

Something like:

```jsx
<IconButton icon={ <Dashicon icon="whatever" style={ { fill: 'red' } } /> } />
```

Reply:
Thank you for this observation!
I didn't consider that option, but it does simplifies the PR quite a bit.
There's a new PR that I have created to replace this one: https://github.com/WordPress/gutenberg/pull/14631
Would you mind to take a look at it? 

Thank you!